Python/ast.c

```diff
@@ -4270,6 +4271,58 @@ decode_bytes_with_escapes(struct compiling *c, const node *n, const char *s,
     return result;
 }
 
+/* Forward declaration for a recursive function. */
+static void fstring_shift_node_locations(node *n, int lineno, int col_offset);
```

Comment:
This line isn't needed, is it? You define the function immediately after the forward declaration.